frontend/src/components/forms/EditResident.tsx

```diff
@@ -0,0 +1,314 @@
+    if (
+      initials.length !== 2 ||
+      moveInDateError || 
+      roomNumberError ||
+      userBuilding === "" ||
+      moveOutDateError
+    ) {
+      return;
+    }
+    editRes();
+    setShowAlert(true);
+  };
+
+  // Timer to remove alert
+  useEffect(() => {
+    if (showAlert) {
+      setTimeout(() => {
+        setShowAlert(false);
+      }, 3000);
+    }
+  }, [showAlert]);
+
+  return (
+    <>
+      <Box>
+        <Modal isOpen={isOpen} onClose={toggleClose} size="xl">
+          <ModalOverlay />
+          <ModalContent>
+            <ModalHeader>Edit Resident</ModalHeader>
+            <ModalCloseButton size="lg" />
+            <ModalBody>
+              <Divider />
+              <Row style={{ marginTop: "16px" }}>
+                <Col>
+                  <FormControl isRequired isInvalid={initialsError}>
+                    <FormLabel>Resident Initials</FormLabel>
+                    <Input
+                      placeholder={initial}
+                      value={initials}
+                      onChange={handleInitialsChange}
+                    />
+                    <FormErrorMessage>
+                      Resident Initials are required and must contain 2 letters.
+                    </FormErrorMessage>
+                  </FormControl>
+                </Col>
+                <Col>
+                  <FormControl isRequired isInvalid={roomNumberError}>
+                    <FormLabel>Room Number</FormLabel>
+                    <Input
+                      value={roomNumber}
+                      onChange={handleRoomNumberChange}
+                      type="number"
+                    />
+                    <FormErrorMessage>
+                      Room Number is required and must only contain numbers.
+                    </FormErrorMessage>
+                  </FormControl>
+                </Col>
+              </Row>
+              <Row style={{ marginTop: "16px" }}>
+                <Col>
+                  <FormControl isRequired isInvalid={moveInDateError}>
+                    <FormLabel>Move In Date</FormLabel>
+                    <SingleDatepicker
+                      name="date-input"
+                      date={moveInDate}
+                      onDateChange={handleMoveInDateChange}
+                      propsConfigs={singleDatePickerStyle}
+                    />
+                    <FormErrorMessage>
+                      Move In Date is required and must be before Move Out Date.
+                    </FormErrorMessage>
+                  </FormControl>
+                </Col>
+              </Row>
+              <Row style={{ marginTop: "16px", marginBottom: "16px" }}>
+                <Col>
+                  <FormControl isRequired isInvalid={buildingError}>
+                    <FormLabel>Building</FormLabel>
+                    <Select
+                      options={BUILDINGS}
+                      placeholder={building}
```

Comment:
we should avoid using placeholders for the select and should stick to defaultValue instead. Here's a [SO](https://stackoverflow.com/questions/43495696/how-to-set-a-default-value-in-react-select) post on how to do so